src/System.AppContext/tests/AppContext.Switch.cs

```diff
@@ -0,0 +1,136 @@
+    }
+
+    [Fact]
+    public void EnableSwitchMultipleTimes()
+    {
+        string switchName = GetRandomSwitchName();
+       
+        AppContext.SetSwitch(switchName, true);
+        AppContext.SetSwitch(switchName, true);
+
+        bool isEnabled;
+        Assert.True(AppContext.TryGetSwitch(switchName, out isEnabled) && isEnabled == true);
+    }
+
+    [Fact]
+    public void DisableSwitch()
+    {
+        string switchName = GetRandomSwitchName();
+        
+        AppContext.SetSwitch(switchName, true);
+        bool isEnabled;
+        Assert.True(AppContext.TryGetSwitch(switchName, out isEnabled) && isEnabled == true);
+
+        // Ensure we can keep the value around
+        AppContext.SetSwitch(switchName, false);
+        Assert.True(AppContext.TryGetSwitch(switchName, out isEnabled) && isEnabled == false);
+    }
+
+    [Fact]
+    public void DisableMultipleSwitches()
+    {
+        string switchName1 = GetRandomSwitchName();
+        string switchName2 = GetRandomSwitchName();
+        
+        AppContext.SetSwitch(switchName1, false);
+        AppContext.SetSwitch(switchName2, false);
+
+        bool isEnabled;
+        Assert.True(AppContext.TryGetSwitch(switchName1, out isEnabled) && isEnabled == false);
+        Assert.True(AppContext.TryGetSwitch(switchName2, out isEnabled) && isEnabled == false);
+    }
+
+    [Fact]
+    public void DisableSwitchMultipleTimes()
+    {
+        string switchName = GetRandomSwitchName();
+        
+        AppContext.SetSwitch(switchName, false);
+        AppContext.SetSwitch(switchName, false);
+
+        bool isEnabled;
+        Assert.True(AppContext.TryGetSwitch(switchName, out isEnabled) && isEnabled == false);
+    }
+
+    [Fact]
+    public void TryGetSwitch_SwitchNotDefined()
+    {
+        string switchName = GetRandomSwitchName();
+        
+        bool isEnabled;
+        var exists = AppContext.TryGetSwitch(switchName, out isEnabled);
+        Assert.False(exists);
+    }
+
+    [Fact]
+    public void TryGetSwitch_SwitchDefined()
+    {
+        bool isEnabled, exists;
+        string randomSwitchName = GetRandomSwitchName();
+        
+        // Enable switch
+        AppContext.SetSwitch(randomSwitchName, true);
+
+        // Get value
+        exists = AppContext.TryGetSwitch(randomSwitchName, out isEnabled);
+        Assert.True(exists);
+        Assert.True(isEnabled);
+
+        Assert.True(AppContext.TryGetSwitch(randomSwitchName, out isEnabled) && isEnabled == true);
+
+        // Disable switch
+        AppContext.SetSwitch(randomSwitchName, false);
+
+        // Get value
+        exists = AppContext.TryGetSwitch(randomSwitchName, out isEnabled);
+        Assert.True(exists);
+        Assert.False(isEnabled);
+        Assert.True(AppContext.TryGetSwitch(randomSwitchName, out isEnabled) && isEnabled == false);
+    }
+
+    private static string GetRandomSwitchName()
+    {
+        return "Switch." + randomNumbers.Next(int.MaxValue).ToString();
```

Comment:
Rather than using a random number, how about adding a [CallerLineNumber] argument to the method.  That way it's deterministic, you don't need the static, and you can correlate failures with where they came from.